jsclient/genclients.go

```diff
@@ -0,0 +1,212 @@
+package jsclient
+
+import (
+	"errors"
+	"io/ioutil"
+	"path/filepath"
+	"regexp"
+
+	"github.com/Clever/wag/swagger"
+	"github.com/Clever/wag/templates"
+	"github.com/go-openapi/spec"
+)
+
+// Generate generates a client
+func Generate(modulePath string, s spec.Swagger) error {
+	pkgName, ok := s.Info.Extensions.GetString("x-npm-package")
+	if !ok {
+		return errors.New("Must provide 'x-npm-package' in the 'info' section of the swagger.yml.") // TODO maybe validate this earlier?
+	}
+
+	tmplInfo := clientCodeTemplate{
+		ClassName:   camelCase(s.Info.InfoProps.Title, true),
+		PackageName: pkgName,
+		ServiceName: s.Info.InfoProps.Title,
+		Version:     s.Info.InfoProps.Version,
+		Description: s.Info.InfoProps.Description,
+	}
+
+	for _, path := range swagger.SortedPathItemKeys(s.Paths.Paths) {
+		pathItem := s.Paths.Paths[path]
+		pathItemOps := swagger.PathItemOperations(pathItem)
+		for _, method := range swagger.SortedOperationsKeys(pathItemOps) {
+			op := pathItemOps[method]
+			if op.Deprecated {
+				continue
+			}
+			methodCode, err := methodCode(op, method, s.BasePath, path)
+			if err != nil {
+				return err
+			}
+			tmplInfo.Methods = append(tmplInfo.Methods, methodCode)
+		}
+	}
+
+	indexJS, err := templates.WriteTemplate(indexJSTmplStr, tmplInfo)
+	if err != nil {
+		return err
+	}
+
+	packageJSON, err := templates.WriteTemplate(packageJSONTmplStr, tmplInfo)
+	if err != nil {
+		return err
+	}
+
+	if err = ioutil.WriteFile(filepath.Join(modulePath, "index.ts"), []byte(indexJS), 0700); err != nil {
```

Comment:
I'd use js file extensions, otherwise all consumers need to be running the ts transpiler